### PR TITLE
Create release v1.6.0

### PR DESCRIPTION
**Pull request by sherryhli**
## Notion ticket link

N/A

## Implementation description
* Followed [release instructions](https://github.com/uwblueprint/starter-code-v2#creating-a-release)
* Includes 3 commits from `main`: 7f803436b8fd2db1c12c56c12f95822b6156b128, 45b76fb13962688ba6568e141b46a50a7bbf2621, de7811552f7798eaf68cfc8f427ddbf5d0c2aa3a
* Fully integrates user registration (sign up) into the release branch

## Steps to test
1. Checkout https://github.com/uwblueprint/starter-code-cli/pull/27
2. For the below stack combinations: generate starter-code using the CLI

**Stacks:** (changes are confined to the controller layer and frontend)
* TypeScript, REST, MongoDB, auth, no-file-storage
* TypeScript, GraphQL, MongoDB, auth, no-file-storage
* Python, REST, MongoDB, auth, no-file-storage

**General testing workflow:**
```bash
# run starter-code-cli locally
# when prompted for a directory, give the path to the directory 1 level above your local starter-code-v2
yarn dev --testing

# run the application
docker-compose up --build

# after testing, reset starter-code-v2
git reset --hard HEAD
```

3. Run the application with `docker-compose up --build`, switch the MG_DATABASE_URL DB between `typescript-test` and `python-test` in .env accordingly

4. Navigate to the frontend, fill out the sign up form with your personal email address (uwblueprint.org already exists as user)
5. Verify that you receive a verification email
6. Before verifying your email, check that the "Display Entities" page returns an error, (we don't handle errors gracefully at the moment)
7. Verify your email and visit "Display Entities" again, you should see data in the table
8. Log out and log in again with the new account, check "Display Entities" page
9. Try creating a user through the `/users` endpoint or `createUser` mutation (you will need an Admin token, authenticate with uwblueprint email), check that a verification email is received

## What should reviewers focus on?
* Take a look at any file that you've touched recently and check to see if anything's out of place
* Are the combinations tested sufficient?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/services/implementations/auth_service.py [hunk 1]
```diff
@@ -102,7 +134,8 @@ def is_authorized_by_role(self, access_token, roles):
             user_role = self.user_service.get_user_role_by_auth_id(
                 decoded_id_token["uid"]
             )
-            return user_role in roles
+            firebase_user = firebase_admin.auth.get_user(decoded_id_token["uid"])
+            return firebase_user.email_verified and user_role in roles
```
Comment (sherryhli) [inline backend/python/app/services/implementations/auth_service.py:138]: Minor syntax issue: this was incorrectly accessed as `firebase_user["email_verified"]` in `main`, fixed for the release and will make a separate PR to fix in `main`

Comment (sherryhli) [reply backend/python/app/services/implementations/auth_service.py:138]: PR for main, please approve as well https://github.com/uwblueprint/starter-code-v2/pull/123

**Review by alexguo8 — APPROVED** [review]
LGTM, don't think there were any new tag additions in the backend, the ones in the frontend look good

**Review by zm2he — APPROVED** [review]
